[IMP] use standard access rights

huguesdk opened the pull request:
*   replace custom access rights by standard groups.
*   use `hr.group_hr_user` instead of `hr.group_hr_manager` for full access to `resource.overtime` and `resource.overtime.rate`.
*   move the overtime rate menuitem under an overtime menuitem to allow for overtime sub-menuitems.

this is related to coopiteasy/cie-custom#60.

huguesdk committed:
[IMP] use standard access rights

* replace custom access rights by standard groups.
* use hr.group_hr_user instead of hr.group_hr_manager for full access to
  resource.overtime and resource.overtime.rate.

huguesdk committed:
[IMP] allow overtime sub-menuitems

move the overtime rate menuitem under an overtime menuitem.

carmenbianca commented on hr_timesheet_overtime/models/hr_employee.py:
```diff
@@ -24,11 +22,13 @@ class HrEmployee(models.Model):
         string="Total Overtime",
         compute="_compute_total_overtime",
         help="Total Overtime since Overtime Start Date",
+        groups="hr.group_hr_user",
```
I'm still confused by how Odoo permissions work, but I think this line is undesireable. It means that users (who are not HR) cannot view their own overtime.

huguesdk replied:
you’re right! fixed! i got confused during my tests: related fields bypass model access rights, but apparently not field access rights. thanks and sorry for the confusion caused on #16.

carmenbianca replied:
Thanks!

huguesdk committed:
[FIX] give access to employees' own total overtime

huguesdk committed:
[FIX] fix pre-commit errors

huguesdk commented:
i’ve just spotted a problem: a regular user (user demo on a new database with demo data) cannot open employee form views due to an access error: (Document type: Employee, Operation: read) - (User: 6, Fields: overtime_start_date). there are many employee fields with groups restrictions and it does not cause an error. why is this one causing it?

huguesdk committed:
[FIX] remove access rights on overtime fields

remove groups restritions on employee overtime fields to avoid access
errors when opening an employee's form.

huguesdk commented:
i've removed access restrictions on the fields altogether since they are not really needed and they cause access errors when opening an employee's form view.

remytms commented:
/ocabot merge patch
